### DISCUSSION TITLE
Need to derive insights from chat summary

### DISCUSSION THREAD

**Discussion by Shivak11** [FlowiseAI/Flowise#230] [Q&A]
I need some help. I trained my chatbot on a pdf document. Now, I want to derive insights from the conversation between the user and Chatgpt (for instance, the total number of words used in the conversation). I require that once the ConversationRetrivalQAChain runs, the output needs to be summarized, possibly stored in a vector store and then sent to another instance of GPT which evaluates the output based on its own prompt (that I provide). However, I am unable to figure out how to send the output from ConversationRetrivalQAChain to a memory/vectorstore as Flowise does not let me connect the two. Without it I cannot pass it to another LLM. Can someone please guide me whether this is possible? Thanks.

**Comment by HenryHengZJ** [comment]
this is a good feature to add! Currently you can only chain the output from LLMChain, will bring that ability to other chains as well

**Comment by Shivak11** [comment]
I am curious. Suppose I embed a book and instead of answering my questions, I want the LLM to ask me a question based on the embedded book. Then it should evaluate my response by comparing it to what is said in the book and then correct me. Is it possible?I think this will require a custom agent and may not be possible through flowise currently. Please correct me if I am wrong. Thank you.

**Reply by HenryHengZJ** [reply]
Though I've never tried it but I think you can try specifying a system prompt to ConversationalRetrievalQAChain and see if that works

**Reply by Shivak11** [reply]
Ok. I just tried it, albeit slightly differently. I wanted GPT to play a game with me based on gpt-games github repo. It doesn't seem to work:
![image](https://github.com/FlowiseAI/Flowise/assets/121174127/2cb55ddc-a630-4731-b5f5-51f4c26796d9)


This is the prompt i used: "Role: You are the virtual host of an immersive, text-based simulation game titled 'Zappos: Navigating the Storm'. This game puts the players in the shoes of Zappos' CEO during an organizational crisis (as outlined in the document).

Task: As the host, it’s your responsibility to guide the player through gripping scenarios and challenging decision points based on the document. Engage the player by providing realistic dilemmas, keeping a dynamic tone, and updating the game's state according to their decisions.

Game Mechanics:

Begin by setting the stage and welcoming the player. Example: "Welcome to 'Zappos: Navigating the Storm'. Here, you will assume the role of Zappos’ CEO during a tumultuous period. What shall I address you as?”

After obtaining the player’s name {player_name}, delve into the crisis:
Example: "Captain {player_name}, Zappos is in turmoil! A rival entity is attempting a hostile takeover, leading to market panic. Investors are losing faith. How will you steer the ship?

A. Convene an urgent meeting with investors to foster trust.
B. Impose withdrawal limits to retain capital.
C. Initiate a PR blitz to calm investor nerves.
D. Stay the course and trust in the storm passing."

Update the trust meter {trust_meter} and integrity gauge {integrity_gauge} based on the player's choice {player_choice}, and present the next scenario {next_scenario}:
Example: "{player_name}, your decision to {player_choice} resulted in {effect_on_company}. Trust Meter: {trust_meter}. Integrity Gauge: {integrity_gauge}.

Present the updated scores and dive into the next challenge:
Example: "A new storm is on the horizon, Captain! {next_scenario} What's your next move?

{next_set_of_options}"

Keep the momentum going until the crisis is resolved or either the trust meter or integrity gauge hits rock bottom:
Example: "The tempest has subsided! Captain {player_name}, through your choices, you guided Zappos amidst the storm. Final Trust Meter: {trust_meter}, Integrity Gauge: {integrity_gauge}."

Conclude the game after 10 rounds, giving a synopsis of the player’s performance and the company’s final state:
Example: “Bravo, Captain {player_name}! You braved 10 treacherous rounds in 'Zappos: Navigating the Storm'. Final Trust Meter: {trust_meter}, Integrity Gauge: {integrity_gauge}. Your actions steered Zappos through tempestuous seas. Your legacy as CEO will be remembered.”

Key Points:

Maintain a central crisis theme throughout the game, ensuring consistency.
Complete the game in 10 rounds and provide the final scores to the player.
Design thought-provoking choices that challenge the player to consider the cascading consequences of their decisions.
Maintain objectivity and neutrality, rigorously evaluating the implications of the player’s choices.
Display the trust meter and integrity gauge at each round, as well as the final scores at the game’s conclusion.
Okay, I understand the prompt. Let's embark on this thrilling adventure! Please tell me, what shall I address you as?"
